maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/AbstractBndMavenPlugin.java

```diff
@@ -649,6 +664,25 @@ protected void reportErrorsAndWarnings(Builder builder) throws MojoFailureExcept
 		}
 	}
 
+	protected boolean isEmpty(File directory) {
+		if (directory == null || !directory.exists()) {
+			return true;
+		}
+
+		Path path = directory.toPath();
+		if (Files.isDirectory(path)) {
+			try (Stream<Path> entries = Files.list(path)) {
```

Comment:
Should this test care about empty directories? That is, what if `directory` contains a tree of directories but zero files? What is the definition of "empty"?

Reply:
what consists of an empty jar?

Reply:
I would say that a non-empty directory must have at least one file in the tree. Just a tree of empty directories is an empty directory. As for a jar, an empty jar is the same as an empty directory with the addition that `META-INF/MANIFEST.MF` does not count as a file in the tree. That is, if the only file in the jar is `META-INF/MANIFEST.MF`, then the jar is empty.